src/rollup/index.ts

```diff
@@ -239,46 +247,67 @@ export interface OutputChunk {
 
 	generate: (outputOptions: OutputOptions) => Promise<{ code: string; map: SourceMap }>;
 	write: (options: OutputOptions) => Promise<void>;
+	getTimings?: () => SerializedTimings;
+}
+
+function applyOptionHook(inputOptions: InputOptions, plugin: Plugin) {
+	if (plugin.options) return plugin.options(inputOptions) || inputOptions;
+	return inputOptions;
+}
+
+function getInputOptions(rawInputOptions: GenericConfigObject): any {
+	if (!rawInputOptions) {
+		throw new Error('You must supply an options object to rollup');
+	}
+	const { inputOptions, deprecations, optionError } = mergeOptions({
+		config: rawInputOptions,
+		deprecateConfig: { input: true }
+	});
+
+	if (optionError) inputOptions.onwarn({ message: optionError, code: 'UNKNOWN_OPTION' });
+	if (deprecations.length) addDeprecations(deprecations, inputOptions.onwarn);
+
+	checkInputOptions(inputOptions);
+	inputOptions.plugins = ensureArray(inputOptions.plugins);
+	return inputOptions.plugins.reduce(applyOptionHook, inputOptions);
 }
 
 export interface OutputChunkSet {
 	chunks: {
 		[chunkName: string]: {
-			name: string,
-			imports: string[],
-			exports: string[],
-			modules: ModuleJSON[]
-		}
+			name: string;
+			imports: string[];
+			exports: string[];
+			modules: ModuleJSON[];
+		};
 	};
 	generate: (outputOptions: OutputOptions) => Promise<{ [chunkName: string]: SourceDescription }>;
 	write: (options: OutputOptions) => Promise<void>;
+	getTimings?: () => SerializedTimings;
 }
 
-export default function rollup (rawInputOptions: InputOptions): Promise<OutputChunk | OutputChunkSet>;
-export default function rollup (rawInputOptions: GenericConfigObject): Promise<OutputChunk | OutputChunkSet> {
-	try {
-		if (!rawInputOptions) {
-			throw new Error('You must supply an options object to rollup');
-		}
-		const { inputOptions, deprecations, optionError } = mergeOptions({
-			config: rawInputOptions,
-			deprecateConfig: { input: true }
-		});
-
-		if (optionError) inputOptions.onwarn({ message: optionError, code: 'UNKNOWN_OPTION' });
+const TIME_BUILD = '# BUILD';
+const TIME_GENERATE = '# GENERATE';
```

Comment:
Ah I see you included the markers to distinguish hierarchy at this level.

Perhaps `timeStart` can take a second argument representing the heading number?